client/landing/stepper/declarative-flow/internals/components/survery-manager/index.tsx

```diff
@@ -1,70 +1,29 @@
-import { useIsEnglishLocale } from '@automattic/i18n-utils';
 import { SITE_MIGRATION_FLOW, HOSTED_SITE_MIGRATION_FLOW } from '@automattic/onboarding';
-import { Suspense } from 'react';
-import { useGeoLocationQuery } from 'calypso/data/geo/use-geolocation-query';
+import { type ComponentType, Suspense } from 'react';
 import AsyncMigrationSurvey from '../../steps-repository/components/migration-survey/async';
 import { Flow } from '../../types';
 import { DeferredRender } from '../deferred-render';
 
-const migrationFlows = [ SITE_MIGRATION_FLOW, HOSTED_SITE_MIGRATION_FLOW ];
-const availableCountries = [ 'US', 'IN' ];
-
-const isMigrationSurveyAvailable = ( {
-	country,
-	flow,
-	isEnglish,
-}: {
-	country?: string;
-	flow?: string;
-	isEnglish?: boolean;
-} ) => {
-	if ( ! country || ! flow || ! isEnglish ) {
-		return false;
-	}
-
-	if ( ! migrationFlows.includes( flow ) ) {
-		return false;
-	}
-
-	if ( ! availableCountries.includes( country ) ) {
-		return false;
-	}
-
-	return true;
-};
-
-const surveyList = [
-	{
-		Component: AsyncMigrationSurvey,
-		isAvailable: isMigrationSurveyAvailable,
-	},
-] as const;
+const FLOW_SURVEYS = new Map< string, ComponentType >( [
+	[ SITE_MIGRATION_FLOW, AsyncMigrationSurvey ],
+	[ HOSTED_SITE_MIGRATION_FLOW, AsyncMigrationSurvey ],
+] );
 
 const SurveyManager = ( { disabled = false, flow }: { disabled?: boolean; flow?: Flow } ) => {
-	const isEnLocale = useIsEnglishLocale();
-	const { data } = useGeoLocationQuery();
-	const countryCode = data?.country_short;
-
-	if ( ! flow || disabled || ! countryCode ) {
+	if ( disabled || ! flow ) {
 		return null;
 	}
 
-	const survey = surveyList.find( ( { isAvailable } ) =>
-		isAvailable( {
-			flow: flow?.variantSlug || flow?.name,
-			country: countryCode,
-			isEnglish: isEnLocale,
-		} )
-	);
+	const SurveyComponent = FLOW_SURVEYS.get( flow.name );
```

Comment:
These changes avoid considering `variantSlug`, but based on my understanding, it shouldn't be needed? Maybe it was included for generalized support of other flows in the future?

Reply:
We can harcode using the `AsyncMigrationSurvey` component, because that's the only one used, isn't it? Maybe the `surveyList` array contained multiple items in the past, but now it has degenerated into just one item. We only need to check if we're inside of one of the two supported flows.

Reply:
Yep, I'd prefer avoiding the indirection here.

Reply:
@jsnajdr By "hard-code", do you envision something different than the changed proposed here, like an inline `if (flow.name !== SITE_MIGRATION_FLOW /* ... */ ) return null;` ?

I interpreted the purpose of this component to be somewhat generic in mapping specific flows to specific surveys. The edits proposed here removed some of the additional considerations for locale/language, but kept that general intent of mapping intact, even if technically it's only supporting one flow currently. I'm all for YAGNI, but I also think this current implementation is straight-forward enough?

Reply:
My $0.02: YAGNI. We might never need that mapping again - I'd suggest removing it and just going with a single survey component.

Reply:
@tyxla @jsnajdr Are the changes in 3b443aa4fa more like what you had in mind?